pkg/pillar/cmd/vaultmgr/vaultmgrzfs.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2018-2019 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package vaultmgr
+
+import (
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	zfsPath              = "/usr/sbin/chroot"
+	defaultZpool         = "persist"
+	defaultSecretDataset = defaultZpool + "/vault"
+)
+
+func getCreateParams(vaultPath string) []string {
+	args := []string{"/hostfs", "zfs", "create", "-o", "encryption=aes-256-gcm", "-o", "keylocation=file://" + zfsHostfsKeyFile, "-o", "keyformat=raw", vaultPath}
+	return args
+}
+
+func getLoadKeyParams(vaultPath string) []string {
+	args := []string{"/hostfs", "zfs", "load-key", vaultPath}
+	return args
+}
+
+func getMountParams(vaultPath string) []string {
+	args := []string{"/hostfs", "zfs", "mount", vaultPath}
+	return args
+}
+
+func unlockZfsVault(vaultPath string, cloudKeyOnlyMode bool) error {
+	//prepare key in the staging file
+	if err := stageKey(cloudKeyOnlyMode, keyDir, keyFile); err != nil {
+		return err
+	}
+	defer unstageKey(keyDir, keyFile)
+
+	//zfs load-key
+	if stdOut, stdErr, err := execCmd(zfsPath, getLoadKeyParams(vaultPath)...); err != nil {
+		log.Errorf("Error loading key for vault: %v, %s, %s", err, stdOut, stdErr)
+		return err
+	}
+	//zfs mount
+	if stdOut, stdErr, err := execCmd(zfsPath, getMountParams(vaultPath)...); err != nil {
+		log.Errorf("Error unlocking vault: %v, %s, %s", err, stdOut, stdErr)
+		return err
+	}
+	return nil
+}
+
+func createZfsVault(vaultPath string, cloudKeyOnlyMode bool) error {
+	//prepare key in the staging file
+	if err := stageKey(cloudKeyOnlyMode, keyDir, keyFile); err != nil {
+		return err
+	}
+	defer unstageKey(keyDir, keyFile)
+	args := getCreateParams(vaultPath)
+	if stdOut, stdErr, err := execCmd(zfsPath, args...); err != nil {
+		log.Errorf("Error creating zfs vault %s, error=%v, %s, %s", vaultPath, err, stdOut, stdErr)
+		return err
+	}
+	log.Infof("Created new vault %s", vaultPath)
+	return nil
+}
+
+func setupZfsVault(vaultPath string) error {
+	keyFile = zfsKeyFile
+	keyDir = zfsKeyDir
```

Comment:
This is confusing ...

Can we not have functions which instead take the dir, file as arguments instead of having a function change some global variables?